rllib/rl_trainers/torch_rl_distributed_trainer.py

```diff
@@ -0,0 +1,115 @@
+        use_gpu = bool(num_workers)
+        scaling_config = ScalingConfig(num_workers=num_workers, use_gpu=use_gpu)
+        queues = [Queue(maxsize=1) for _ in range(num_workers)]
+        backend_config = TorchConfig()
+        config = {"queues": queues, "rl_module_config": config["rl_module_config"], 
+            "rl_module_class": config["rl_module_class"]}
+        train_loop_per_worker = construct_train_func(
+            self.training_func,
+            config,
+            fn_arg_name="train_loop_per_worker",
+            discard_returns=True,
+        )
+        self.debug_mode = config.get("debug_mode", False)
+
+        backend_executor = BackendExecutor(
+                backend_config=backend_config,
+                num_workers=scaling_config.num_workers,
+                num_cpus_per_worker=scaling_config.num_cpus_per_worker,
+                num_gpus_per_worker=scaling_config.num_gpus_per_worker,
+                max_retries=0,
+            )
+
+        checkpoint_manager = CheckpointManager(
+                    checkpoint_strategy=None, run_dir=None
+                )
+
+        # Start the remote actors.
+        backend_executor.start(initialization_hook=None)
+
+        empty_spec = RayDatasetSpec(dataset_or_dict=None)
+
+        self.training_iterator = TrainingIterator(
+            backend_executor=backend_executor,
+            backend_config=backend_config,
+            train_func=train_loop_per_worker,
+            dataset_spec=empty_spec,
+            checkpoint_manager=checkpoint_manager,
+            checkpoint=None,
+            checkpoint_strategy=None,
+        )
+        self.curr_weights = None
+
+    @classmethod
+    def training_func(self, config):
+        if "rl_module_config" not in config:
+            raise ValueError("rl_module_config not in config")
+        rl_module_config = config["rl_module_config"]
+        rl_module_class = config["rl_module_class"]
+        rl_module = rl_module_class(rl_module_config)
+        queue = config["queues"][session.get_local_rank()]
+
+        device = train.torch.get_device()
+        num_models = rl_module.num_models
+
+        pgs = [torch.distributed.new_group(list(range(session.get_world_size()))) for _ in range(num_models)]
+        moved_models = []
+        for pg, model in zip(pgs, rl_module.models):
+            model.to(device)
+            model = DDP(model, device_ids=[session.get_local_rank()], process_group=pg)
+            moved_models.append(model)
+
+        rl_module.set_models(moved_models)
+        rl_module.init_optimizers()
+
+        while 1:
+            data = queue.get()
+            out = []
+            result = {}
+            result["loss"] = {}
+            for model in rl_module.models:
+                # this has to call model.__call__ or model.forward
+                # eventually in order to work with ddp
+                out = model.forward_train(data)
+                loss = model.loss(out)
```

Comment:
Model's are not aware of their losses. RLTrainer would not about the loss functionality.